src/execution_plan/ops/op_join.c

```diff
@@ -35,12 +36,19 @@ static Record JoinConsume(OpBase *opBase) {
 	OpJoin *op = (OpJoin *)opBase;
 	Record r = NULL;
 
+	bool update_column_map = false;
 	while(!r) {
 		// Try pulling from current stream.
 		r = OpBase_Consume(op->stream);
+		if(update_column_map && r) {
+			// We have a new record mapping, update the ResultSet column map to match it.
+			ResultSet_SetColToRecMap(QueryCtx_GetResultSet(), r);
+			update_column_map = false;
```

Comment:
Can you explain this fix?

Reply:
Since this PR makes RETURN/OpResults part of the previous scope instead of being independent, the various record maps built by the UNION-joined streams may not be aligned. This change causes the mapping inside the ResultSet to be updated once for each joined stream.